From 09b0e8c6a0a4607763c8900193b3cf94132eea01 Mon Sep 17 00:00:00 2001
Message-Id: <09b0e8c6a0a4607763c8900193b3cf94132eea01.1706105572.git.rahul.singh@arm.com>
From: Rahul Singh <rahul.singh@arm.com>
Date: Thu, 11 Jan 2024 15:49:45 +0000
Subject: [PATCH] rss/kronos: Update Tower-NCI to include the SI in RSS ATU
 memory map

On latest FVP, Safety Island is standalone and has its own address space
in RSS ATU memory map. This patch will update Tower-NCI config to
include the Safety Island in RSS ATU memory map to access the SI during
boot. Also, update Safety Island base address to access the Safety
Island resources during runtime.

Upstream-Status: Pending
Signed-off-by: Rahul Singh <rahul.singh@arm.com>
---
 platform/ext/target/arm/rss/kronos/host_base_address.h |  2 +-
 platform/ext/target/arm/rss/kronos/tower_nci_lib.c     | 10 +++++-----
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/platform/ext/target/arm/rss/kronos/host_base_address.h b/platform/ext/target/arm/rss/kronos/host_base_address.h
index 27a147adb..e9465a4a9 100644
--- a/platform/ext/target/arm/rss/kronos/host_base_address.h
+++ b/platform/ext/target/arm/rss/kronos/host_base_address.h
@@ -180,7 +180,7 @@
 /*  SI CL0 Image address start, offset so end of HEADER is at end of ATU HEADER */
 #define HOST_SI_CL0_IMG_BASE_S         (HOST_SI_CL0_CODE_BASE_S - BL2_HEADER_SIZE)
 /*  SI physical address start */
-#define HOST_SI_PHYS_BASE              0x2000000000000ULL
+#define HOST_SI_PHYS_BASE              0x8000000000000ULL
 /*  SI CL0 ATU CODE size (aligned size of SI image) */
 #define HOST_SI_CL0_ATU_SIZE           ALIGN_UP(SIZE_DEF_SI_CL0_IMAGE, RSS_ATU_PAGE_SIZE)
 /*  SI CL0 ID for SI CL0 ATU HEADER region */
diff --git a/platform/ext/target/arm/rss/kronos/tower_nci_lib.c b/platform/ext/target/arm/rss/kronos/tower_nci_lib.c
index ec967155a..9261d2008 100644
--- a/platform/ext/target/arm/rss/kronos/tower_nci_lib.c
+++ b/platform/ext/target/arm/rss/kronos/tower_nci_lib.c
@@ -45,7 +45,7 @@ enum sysctrl_xMNI_ids {
     SYSCTRL_LCP_AMNI_ID,
     SYSCTRL_LCP_SCP_AMNI_ID,
     SYSCTRL_RSM_AMNI_ID,
-    SYSCTRL_RSS_MCP_AMNI_ID,
+    SYSCTRL_RSS_SI_AMNI_ID,
     SYSCTRL_RSS_SCP_AMNI_ID,
     SYSCTRL_CMN_PMNI_ID,
     SYSCTRL_RSM_PMNI_ID,
@@ -264,7 +264,7 @@ uint32_t program_sysctrl_psam(struct tower_nci_discovery_tree_t* dis_tree,
         { mhu_send_frame_rss.base_2,          mhu_send_frame_rss.end_2,           SYSCTRL_APP_AMNI_ID    , 14 },
         { 0x0004000000000,                    0x0FFFFFFFFFFFF,                    SYSCTRL_APP_AMNI_ID    , 15 },
         { 0x1000000000000,                    0x10000FFFFFFFF,                    SYSCTRL_RSS_SCP_AMNI_ID, 16 },
-        { 0x2000000000000,                    0x2000FFFFFFFFF,                    SYSCTRL_RSS_MCP_AMNI_ID, 17 }
+        { 0x8000000000000,                    0x800FFFFFFFFFF,                    SYSCTRL_RSS_SI_AMNI_ID,  17 },
     };
 
     struct psam_region_cfgs scp_axis[] = {
@@ -398,8 +398,8 @@ uint32_t program_sysctrl_apu(struct tower_nci_discovery_tree_t* dis_tree,
         { 0x1000000000000, 0x10000FFFFFFFF, T_NCI_FOREGROUND, T_NCI_ALL_PERM, 0, },
     };
 
-    struct apu_region_cfgs rss_mcp_axim[] = {
-        { 0x2000000000000, 0x2000FFFFFFFFF, T_NCI_FOREGROUND, T_NCI_ALL_PERM, 0, },
+    struct apu_region_cfgs rss_si_axim[] = {
+        { 0x8000000000000, 0x800FFFFFFFFFF, T_NCI_FOREGROUND, T_NCI_ALL_PERM, 0, },
     };
 
     struct apu_cfgs apu_table[] = {
@@ -410,7 +410,7 @@ uint32_t program_sysctrl_apu(struct tower_nci_discovery_tree_t* dis_tree,
         APU_TUPLE(lcp_axim,     TOWER_NCI_AMNI, SYSCTRL_LCP_AMNI_ID),
         APU_TUPLE(lcp_axis,     TOWER_NCI_ASNI, SYSCTRL_LCP_ASNI_ID),
         APU_TUPLE(rss_scp_axim, TOWER_NCI_AMNI, SYSCTRL_RSS_SCP_AMNI_ID),
-        APU_TUPLE(rss_mcp_axim, TOWER_NCI_AMNI, SYSCTRL_RSS_MCP_AMNI_ID),
+        APU_TUPLE(rss_si_axim,  TOWER_NCI_AMNI, SYSCTRL_RSS_SI_AMNI_ID),
     };
 
     if (program_apu_table(apu_table, sizeof(apu_table) /
-- 
2.25.1

